src/Components/Notification.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+interface NotificationPropsType {
+  hasErrorFromServer: boolean,
+  clearNotification: () => void,
+  errorMessage: string,
+}
+
+export const Notification: React.FC<NotificationPropsType> = ({
+  hasErrorFromServer,
+  clearNotification,
+  errorMessage,
+}) => {
+  const [hidden, setHidden] = useState(true);
+
+  useEffect(() => {
+    if (hasErrorFromServer) {
+      setHidden(false);
+      const timerId = window.setTimeout(() => {
+        setHidden(true);
+        clearNotification();
+        window.clearTimeout(timerId);
+      }, 3000);
+    } else {
+      setHidden(true);
+    }
+  }, [hasErrorFromServer]);
```

Comment:
Seems that you forgot to add the clearNotification function to a dependency. Wrap clearNotification into the useCallback as well
```suggestion
  }, [hasErrorFromServer, clearNotification]);
```